tests/conftest.py

```diff
@@ -0,0 +1,85 @@
+from pytest import fixture
+from fastapi.testclient import TestClient
+from sqlmodel.pool import StaticPool
+from sqlmodel import Session, SQLModel, create_engine
+from chowda.app import app
+from chowda.session import get_session
+from factories import (
+    MediaFileFactory,
+    BatchFactory,
+    CollectionFactory,
+    ClamsAppFactory,
+    PipelineFactory,
+    ClamsEventFactory,
+)
+from factory.alchemy import SQLAlchemyModelFactory
+
+
+@fixture(name='engine')
+def engine_fixture():
+    yield create_engine(
+        'sqlite://', connect_args={'check_same_thread': False}, poolclass=StaticPool
+    )
+
+
+@fixture(name='session')
+def session_fixture(engine):
+    SQLModel.metadata.create_all(engine)
+    with Session(engine) as session:
+        yield session
+
+
+@fixture(name='client')
+def client_fixture(session: Session):
+    def get_session_override():
+        return session
+
+    app.dependency_overrides[get_session] = get_session_override
+    client = TestClient(app)
+    yield client
+    app.dependency_overrides.clear()
+
+
+def override_session(factory: SQLAlchemyModelFactory, session: Session):
+    """Override the default factory session with the given session.
+
+    This is necessary because factory_boy sessions are set at class creation,
+    but here each test needs it's own independent session for full isolation.
+    """
+    factory._meta.sqlalchemy_session = session
+
+
+@fixture(name='MediaFileFactory')
+def MediaFileFactory_fixture(session: Session):
+    override_session(MediaFileFactory, session)
+    yield MediaFileFactory
+
+
+@fixture(name='BatchFactory')
+def BatchFactory_fixture(session: Session):
+    override_session(BatchFactory, session)
+    yield BatchFactory
+
+
+@fixture(name='CollectionFactory')
+def CollectionFactory_fixture(session: Session):
+    override_session(CollectionFactory, session)
+    yield CollectionFactory
+
+
+@fixture(name='ClamsAppFactory')
+def ClamsAppFactory_fixture(session: Session):
+    override_session(ClamsAppFactory, session)
+    yield ClamsAppFactory
+
+
+@fixture(name='PipelineFactory')
+def PipelineFactory_fixture(session: Session):
+    override_session(PipelineFactory, session)
+    yield PipelineFactory
+
+
+@fixture(name='ClamsEventFactory')
+def ClamsEventFactory_fixture(session: Session):
+    override_session(ClamsEventFactory, session)
+    yield ClamsEventFactory
```

Comment:
I tried creating these programatically, but wasn't able to get it working. I'm sure this can be simplified, but it's fine as is.